shell_integration/windows/OCOverlays/OCOverlay.cpp

```diff
@@ -126,29 +127,27 @@ IFACEMETHODIMP OCOverlay::GetPriority(int *pPriority)
     return S_OK;
 }
 
- IFACEMETHODIMP OCOverlay::IsMemberOf(PCWSTR pwszPath, DWORD dwAttrib)
+IFACEMETHODIMP OCOverlay::IsMemberOf(PCWSTR pwszPath, DWORD dwAttrib)
 {
     RemotePathChecker* checker = getGlobalChecker();
-    auto watchedDirectories = checker->WatchedDirectories();
+    checker->UpdateWatchedDirectories(_localWatchedDirectories, _localWatchedDirectoriesVersion);
+    if (_localWatchedDirectories.empty())
+        return MAKE_HRESULT(S_FALSE, 0, 0);
 
-    wstring wpath(pwszPath);
-    wpath.append(L"\\");
-    vector<wstring>::iterator it;
     bool watched = false;
-    for (it = watchedDirectories.begin(); it != watchedDirectories.end(); ++it) {
-        if (StringUtil::begins_with(wpath, *it)) {
+    size_t pathLength = wcslen(pwszPath);
+    for (auto it = _localWatchedDirectories.cbegin(); it != _localWatchedDirectories.cend(); ++it) {
+        if (StringUtil::isDescendantOf(pwszPath, pathLength, *it)) {
             watched = true;
         }
     }
 
-    if (!watched) {
+    if (!watched)
         return MAKE_HRESULT(S_FALSE, 0, 0);
-    }
 
     int state = 0;
-    if (!checker->IsMonitoredPath(pwszPath, &state)) {
+    if (!checker->IsMonitoredPath(pwszPath, &state))
```

Comment:
Reverted that change and put a brace above, let's have that discussion officially before we start doing it.